### PR TITLE
Fix bugs in documentation and reenable algolia search

### PR DESCRIPTION
**Pull request by MichaelHoepler**
This is the PR for just the changes in the docs.

### CODE REVIEW

**Comment by frrist** [comment]
@MichaelHoepler feel free to ping me when this is out of draft for a review.

**Comment by MichaelHoepler** [comment]
@frrist Did you take a look at the PR?

File: docs/docs/all-flags.md [hunk 1]
```diff
@@ -111,21 +111,21 @@ An example job in YAML format:
...
+        parameters:
+            - Hello
+            - World
+    outputs:
+        - name: outputs
+          path: /outputs
 deal:
-  concurrency: 1
+    concurrency: 1
 ```
```
Comment (frrist) [inline docs/docs/all-flags.md:129]: yaml is space sensitive, and it looks like you have made modifications here (I assume they were unintentional) please ensure this yaml is still valid and revert this change if its not.

File: docs/docs/all-flags.md [hunk 2]
```diff
@@ -283,7 +283,7 @@ Flags:
       --local                            Run the job locally. Docker is required
       --memory string                    Job Memory requirement (e.g. 500Mb, 2Gb, 8Gb).
       --min-bids int                     Minimum number of bids that must be received before concurrency-many bids will be accepted (at random)
-      --network network-type             Networking capability required by the job (default "nats")
+      --network network-type             Networking capability required by the job (default None)
```
Comment (frrist) [inline docs/docs/all-flags.md:286]: revert this, the default network is nats.

**Review by frrist — COMMENTED** [review]
Leaving an early comment (I am still reviewing) please revert all changes that are only adding trailing whitespace to the end of lines. There are quite a few of these and it's causing the diff of the PR to be larger (and harder to review) than need be.

In general, aim to make modifications to the docs only where necessary to keep the PR small and review easy. Right now it's a little hard to tell what's changed.

File: docs/docs/community/compute-landscape.md [hunk 3]
```diff
@@ -8,7 +8,7 @@ sidebar_position: 2
 This page is an introduction to a landscape analysis of general-purpose compute frameworks.
```
Comment (frrist) [inline docs/docs/community/compute-landscape.md:8]: revert changes to this file.

File: docs/docs/community/style-guide.md [hunk 4]
```diff
@@ -14,14 +14,14 @@ The primary objective of the Bacalhau documentation style guide is to help autho
 
 Our goal is to use conversational tone that is natural, and friendly towards the reader and also ensure that the document's content is simple and easy to follow.
```
Comment (frrist) [inline docs/docs/community/style-guide.md:15]: revert changes to this file

File: docs/docs/dev/api/index.md [hunk 5]
```diff
@@ -4,19 +4,19 @@ Welcome to the official API documentation for Bacalhau. This guide provides a de
```
Comment (frrist) [inline docs/docs/dev/api/index.md:4]: revert changes to this file

File: docs/docs/dev/auth_flow.md [hunk 6]
```diff
@@ -1,36 +1,7 @@
-# Authorization and authentication flow
+# Authorization/authentication flow
 
 Bacalhau authenticates and authorizes users in a multi-step flow.
 
-## Requirements
```
Comment (frrist) [inline docs/docs/dev/auth_flow.md:5]: What is the motivation for changing this file? A lot of this content was helpful, why is it being removed?

File: docs/docs/dev/other-specifications/engines/docker.md [hunk 7]
```diff
@@ -39,3 +39,8 @@ Engine:
 ```
 
 In this example, the task will be executed inside an Ubuntu 20.04 Docker container. The entrypoint is overridden to execute a bash shell that runs an echo command. An environment variable MY_ENV_VAR is set with the value myvalue, and the working directory inside the container is set to /app.
+
+## Notes
+
+- Currently, only images published to public registries are supported. Make sure your image is available in one of them, for example DockerHub or ghcr. Private images can be retrieved from docker hub by [specifying your credentials](../../../setting-up/running-node/quick-start-docker.md#authenticate-with-docker-hub).
+- Make sure that the architecture of your image matches the architecture of the nodes in your cluster. Also consider the possibility of using [multi-platform](https://docs.docker.com/build/building/multi-platform/) images.
```
Comment (frrist) [inline docs/docs/dev/other-specifications/engines/docker.md:46]: > Currently, only images published to public registries are supported

I don't think this is true. The docker daemon attached to the compute node can pull containers it has credentials to access. 

> Private images can be retrieved from docker hub by ...

Okay now we walk it back, this is correct, we can fetch private images if credentials are provided

---------------------------------------------

Please fix this section, its confusing.

File: docs/docs/dev/other-specifications/publishers/ipfs.md [hunk 8]
```diff
@@ -29,4 +32,4 @@ PublishedResult:
     CID: "QmXoypizjW3WknFiJnKLwHCnL72vedxjQkDDP1mXWo6uco"
 ```
 
-In this example, the task results will be stored in IPFS, and can be referenced and retrieved using the specified CID. This is indicative of Bacalhau's commitment to offering flexible, reliable, and decentralized options for result storage, catering to a diverse set of user needs and preferences.
+In this example, the task results will be stored in IPFS, and can be referenced and retrieved using the specified CID. This is indicative of Bacalhau's commitment to offering flexible, reliable, and decentralized options for result storage, catering to a diverse set of user needs and preferences.
```
Comment (frrist) [inline docs/docs/dev/other-specifications/publishers/ipfs.md:35]: What has changed here?

File: docs/docs/dev/other-specifications/publishers/s3.md [hunk 9]
```diff
@@ -13,7 +13,11 @@ Bacalhau's S3 Publisher provides users with a secure and efficient method to pub
 - **Key** `(string: <required>)`: The object key within the specified bucket where the task results will be stored.
 - **Endpoint** `(string: <optional>)`: The endpoint URL of the S3 service (useful for S3-compatible services).
 - **Region** `(string: <optional>)`: The region where the S3 bucket is located.
+- **Compress** `(bool: false)`: Indicates whether the task results should be compressed before storage. Only compressed objects can be downloaded using the `bacalhau get <job_id>` command.
 
+## Publishing Flexibility
+
+The S3 Publisher is adept at handling both individual files and full directories. Each file within a directory is uploaded as a separate S3 object. If the `Compress` option is enabled, the entire directory is compressed into a single object, enhancing efficiency and reducing storage requirements. Only compressed objects can be downloaded using the `bacalhau get <job_id>` command.
```
Comment (frrist) [inline docs/docs/dev/other-specifications/publishers/s3.md:20]: @wdbaruni can you comment if this is still true? If I remember correctly we always compress now.

File: docs/docs/dev/other-specifications/publishers/s3.md [hunk 10]
```diff
@@ -26,6 +30,10 @@ Results published to S3 are stored as objects that can also be used as inputs to
 - **VersionID**: The version ID of the stored object, enabling versioning support for retrieving specific versions of stored data.
 - **ChecksumSHA256**: The SHA-256 checksum of the stored object, providing a method to verify data integrity.
 
+**Note:** 
+- `ChecksumSHA256` and `VersionID` are only returned when the `Compress` option is enabled, offering users a method to verify the integrity of the compressed data and to track different versions of the stored objects.
+- `ChecksumSHA256` is set only when working with AWS S3, but is not set for all other S3-compatible services.
+
```
Comment (frrist) [inline docs/docs/dev/other-specifications/publishers/s3.md:36]: @wdbaruni to comment here as well.

File: docs/docs/dev/other-specifications/publishers/s3.md (hunk 9 above)
Comment (frrist) [reply docs/docs/dev/other-specifications/publishers/s3.md:20]: Compress is not part of the spec: 
https://github.com/bacalhau-project/bacalhau/blob/main/pkg/s3/types.go#L49

File: docs/docs/dev/other-specifications/publishers/s3.md (hunk 10 above)
Comment (frrist) [reply docs/docs/dev/other-specifications/publishers/s3.md:36]: Since we always compress I think these are always set?

File: docs/docs/dev/other-specifications/publishers/s3.md [hunk 11]
```diff
@@ -70,6 +78,7 @@ Publisher:
     Bucket: "my-task-results"
     Key: "task123/result.tar.gz"
     Endpoint: "https://s3.us-west-2.amazonaws.com"
+    Compress: true
```
Comment (frrist) [inline docs/docs/dev/other-specifications/publishers/s3.md:81]: remove

File: docs/docs/dev/other-specifications/publishers/s3.md [hunk 12]
```diff
@@ -95,15 +104,21 @@ The Bacalhau command-line interface (CLI) provides an imperative approach to spe
...
 2. **Docker job writing to S3 with a specific endpoint and region**:
    ```bash
    bacalhau docker run -p s3://bucket/key,opt=endpoint=http://s3.example.com,opt=region=us-east-1 ubuntu ...
    ```
    This command specifies a unique endpoint and region for the S3 bucket.
 
-3. **Using naming placeholders**:
+3. **Docker job writing a single archived file to S3**:
+   ```bash
+   bacalhau docker run -p s3://bucket/key,opt=compress=true ubuntu ...
```
Comment (frrist) [inline docs/docs/dev/other-specifications/publishers/s3.md:117]: cc @wdbaruni I believe we remove compress as an option.

File: docs/docs/dev/other-specifications/sources/ipfs.md [hunk 13]
```diff
@@ -41,4 +41,4 @@ Utilizing IPFS as an input source in Bacalhau via the CLI is straightforward. Be
    bacalhau docker run -i ipfs://QmeZRGhe4PmjctYVSVHuEiA9oSXnqmYa4kQubSHgWbjv72:/data ubuntu ...
    ```
 
-These commands provide a seamless mechanism to fetch and mount data from IPFS directly into your task's execution environment using the Bacalhau CLI.
```
Comment (frrist) [inline docs/docs/dev/other-specifications/sources/ipfs.md:44]: revert this change as it appears nothing changed

File: docs/docs/dev/other-specifications/sources/s3.md [hunk 14]
```diff
@@ -12,7 +12,7 @@ Here are the parameters that you can define for an S3 input source:
 
 - **Bucket** `(string: <required>)`: The name of the S3 bucket where the data is stored.
```
Comment (frrist) [inline docs/docs/dev/other-specifications/sources/s3.md:13]: revert this change

File: docs/docs/dev/setting_up_development.md [hunk 15]
```diff
@@ -1,17 +1,16 @@
...
 
 **Instructions**
-
 - Set environment variables:
-
 ```
 export PYTHONVER='3.11.7'
 export GOLANGCILINTVER='v1.51.2'
-export GOLANGVER='1.21'
+export GOLANGVER='1.20'
```
Comment (frrist) [inline docs/docs/dev/setting_up_development.md:13]: revert this change. We are using 1.21

**Review by frrist — COMMENTED** [review]
I am going to pause on reviewing this further. The current PR modifies over 70% of the files making up our docs. It is unreasonable to expect a complete review of a PR this size. Please consider taking the following steps to make this work reviewable:
- Revert all changes which only modify white space.
- Separate the work done here into multiple PRs with clear descriptions and motivation for the changes they are making.
- Separate PRs that change document content from PRs that change document styling.

I am all for making our docs better, but we need to approach making these changes with care, and we can do that by breaking this down into reviewable pieces of work.

On the other hand, if you are confident the changes here are correct and would prefer to merge this now and fail forward - making adjustments in follow on PRs - I am good with that too.
